sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/AssemblyReference.cs

```diff
@@ -87,24 +91,35 @@ MetadataReference[] ProcessNuGet()
                 packageManager.InstallPackage(packageId, SemanticVersion.ParseOptionalVersion(packageVersion),
                     ignoreDependencies: true, allowPrereleaseVersions: false);
 
-                var matchingDlls = Directory.GetFiles(GetNuGetPackageDirectory(), "*.dll", SearchOption.AllDirectories)
-                    .Select(path => new FileInfo(path))
-                    .GroupBy(file => file.Directory.Name)
-                    .Where(group => allowedNugetLibDirectories.Any(y => group.Key.StartsWith(y)))
-                    .OrderByDescending(group => group.Key)
-                    .First();
-
-                return matchingDlls.Select(file => MetadataReference.CreateFromFile(file.FullName)).ToArray();
+                return GetFilesFromLatestFrameworkVersion(packageId, packageVersion);
```

Comment:
No functional changes, just refactored with additional comments to make it clear what is being done.